src/configure/templateInputHelper/RepoAnalysisSettingInputProvider.ts

```diff
@@ -71,10 +88,16 @@ export class RepoAnalysisSettingInputProvider {
                 else {
                     selectedValue = await new ControlProvider().showQuickPick(repoAnalysisSettingKey, possibleValues, { placeHolder: inputControl.getInputDescriptor().name });
                 }
-
                 if (settingIndexMap.get(selectedValue.data).length === 1) {
                     this._selectedRepoAnalysisSettingIndex = settingIndexMap.get(selectedValue.data)[0];
                 }
+
+                let repoAnalysisValue = this._repoAnalysisSettings[settingIndexMap.get(selectedValue.data)[0]].settings[repoAnalysisSettingKey];
```

Comment:
shouldn't this be done earlier when we are showing the drop down at L89? Why do you want to show the drop down twice?

Reply:
@bishal-pdMSFT this is the case when there multiple repository analysis setting and the setting key value is an array and the value for that setting key to be used is string. So here first dropdown allows the value to be picked from various repository analysis settings and second dropdown is to get the value for the setting.